coderd/telemetry/telemetry.go

```diff
@@ -287,7 +287,9 @@ func (r *remoteReporter) deployment() error {
 // createSnapshot collects a full snapshot from the database.
 func (r *remoteReporter) createSnapshot() (*Snapshot, error) {
 	var (
-		ctx = r.ctx
+		// Requires read access to all resources.
+		//nolint:gocritic
+		ctx = dbauthz.AsSystemRestricted(r.ctx)
```

Comment:
Why does this need to use authz at all? This seems like additional abstraction without much benefit, since we know it needs everything.

Reply:
I see that this is actually not launched from `coderd.New`, so I don't think the database passed to telemetry has any dbauthz in front.

I was just doing this to be consistent, since all access to the database should be checked, in all locations of the code. But this actually sidesteps dbauthz anyway right now.